core/src/main/java/tc/oc/pgm/channels/MessageChannel.java

```diff
@@ -0,0 +1,292 @@
+import tc.oc.pgm.api.integration.Integration;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.setting.SettingKey;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.util.MessageSenderIdentity;
+import tc.oc.pgm.util.Players;
+import tc.oc.pgm.util.bukkit.OnlinePlayerMapAdapter;
+import tc.oc.pgm.util.named.NameStyle;
+
+public class MessageChannel implements Channel<MatchPlayer> {
+
+  private final ChannelManager manager;
+  private final OnlinePlayerMapAdapter<MessageSenderIdentity> setChannel, lastMessagedBy;
+
+  private static final Sound SOUND = sound(key("random.orb"), Sound.Source.MASTER, 1f, 1.2f);
+
+  public MessageChannel(ChannelManager manager) {
+    this.manager = manager;
+    this.setChannel = new OnlinePlayerMapAdapter<MessageSenderIdentity>(PGM.get());
+    this.lastMessagedBy = new OnlinePlayerMapAdapter<MessageSenderIdentity>(PGM.get());
+  }
+
+  @Override
+  public String getDisplayName() {
+    return "messages";
+  }
+
+  @Override
+  public String[] getAliases() {
+    return new String[] {"msg", "tell", "r"};
+  }
+
+  @Override
+  public Character getShortcut() {
+    return '@';
+  }
+
+  @Override
+  public MatchPlayer getTarget(MatchPlayer sender, Map<String, ?> arguments) {
+    MatchPlayer target = (MatchPlayer) arguments.get("target");
+    if (target == null) {
+      return getSelectedTarget(sender);
+    } else {
+      checkSettings(target, sender);
+
+      if (!target.equals(getSelectedTarget(sender))
+          && !(manager.getSelectedChannel(sender) instanceof MessageChannel)) {
+        this.lastMessagedBy.put(
+            sender.getBukkit(), new MessageSenderIdentity(sender.getBukkit(), target.getBukkit()));
+      }
+
+      if (!sender.equals(getSelectedTarget(target))
+          && !(manager.getSelectedChannel(target) instanceof MessageChannel)) {
+        this.lastMessagedBy.put(
+            target.getBukkit(), new MessageSenderIdentity(target.getBukkit(), sender.getBukkit()));
+      }
+
+      return target;
+    }
+  }
+
+  @Override
+  public Collection<MatchPlayer> getViewers(MatchPlayer target) {
+    if (target == null) throw exception("command.playerNotFound");
+    return Collections.singletonList(target);
+  }
+
+  @Override
+  public void sendMessage(ChannelMessageEvent<MatchPlayer> event) {
+    MatchPlayer sender = event.getSender();
+    MatchPlayer target = event.getTarget();
+
+    SettingValue value = target.getSettings().getValue(SettingKey.SOUNDS);
+    if (value.equals(SettingValue.SOUNDS_ALL)
+        || value.equals(SettingValue.SOUNDS_CHAT)
+        || value.equals(SettingValue.SOUNDS_DM)) target.playSound(SOUND);
+
+    Bukkit.getPluginManager()
+        .callEvent(
+            new ChannelMessageEvent<MatchPlayer>(
+                event.getChannel(),
+                sender,
+                event.getTarget(),
+                Collections.singletonList(sender),
+                event.getMessage()));
```

Comment:
no, sending one message should not trigger a separate message event

it simply should be that sending itself happens within this method, and it just so happens that for DMs one message sends two chat messages (one with "From" and one with "To"). This is fine: in global chat one message sends N messages (to N recipients) with just one event.